src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1500,6 +1502,14 @@ public function getContent(bool $asResource = false)
         return $this->content;
     }
 
+    /**
+     * Get the application/x-form-urlencoded or application/json request body.
+     */
+    public function getPayload(): InputBag
+    {
+        return $this->payload ??= $this->request->count() ? $this->request : new InputBag($this->toArray());
```

Comment:
im wondering if we should throw Validation/HttpException from `toArray()` :thinking:

Reply:
That's already the case: JsonException extends RequestExceptionInterface, which leads to 400.